core/src/main/java/io/undertow/server/handlers/SameSiteCookieHandler.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.undertow.server.handlers;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.jboss.logging.Logger;
+import io.undertow.UndertowMessages;
+import io.undertow.server.HandlerWrapper;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.server.ResponseCommitListener;
+import io.undertow.server.handlers.builder.HandlerBuilder;
+import io.undertow.util.Headers;
+import io.undertow.util.SameSiteNoneIncompatibleClientChecker;
+
+/**
+ * Handler that will set the SameSite flag on all cookies
+ */
+public class SameSiteCookieHandler implements HttpHandler {
+
+    private static final Logger log = Logger.getLogger(SameSiteCookieHandler.class);
+
+    private final HttpHandler next;
+    private final String mode;
+    private final String cookieName;
+    private final boolean disableClientChecker;
+
+    public SameSiteCookieHandler(final HttpHandler next, final String mode) {
+        this(next, mode, null, false);
+    }
+
+    public SameSiteCookieHandler(final HttpHandler next, final String mode, final String cookieName) {
+        this(next, mode, cookieName, false);
+    }
+
+    public SameSiteCookieHandler(final HttpHandler next, final String mode, final String cookieName, final boolean disableClientChecker) {
+        this.next = next;
+        this.mode = mode;
+        this.cookieName = cookieName;
```

Comment:
It would be ideal with this cookie name were a pattern

Reply:
Done. I've changed the parameter from "cookie-name" to "cookie-pattern" which can support regex. Also, I've added a new parameter "case-sensitive" (true by default) which can support incase-sensitive match by setting it to false.